working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1748,65 +1746,196 @@ types of all of the case expressions.
+    and does not bind any new ones.
+
+*   **Switch statement**, **switch expression**, **if-case statement**: Each
+    `caseHead` introduces a new *case scope* which is where the variables
+    defined by that case's pattern are bound.
+
+    There is no *initializing expression* for the variables in a case pattern,
+    but they are considered initialized after the entire case pattern, before
+    the guard expression if there is one. *However, all pattern variables are
+    in scope in the entire pattern:*
+
+    ```dart
+    const c = 1;
+    switch (1) {
+      case [var c, == c]
+        //            ^ Error: In scope but not initialized.
+        //              (Also an error because `c` is not a constant.)
+            when c == 2:
+        //       ^ OK.
+        print(c);
+        //    ^ OK.
+    }
+    ```
 
-The scope where a pattern's variables are declared and available for use depends
-on the construct that contains the pattern:
+    The guard expression is evaluated in its case's case scope.
 
-*   **Local pattern variable declaration**: The rest of the block following
-    the declaration.
-*   **For loop pattern variable declaration**: The body of the loop and the
-    condition and increment clauses in a C-style for loop.
-*   **Switch statement case**: The guard clause and the statements of the
-    subsequent non-empty case body.
-*   **Switch expression case**: The guard clause and the case expression.
-*   **If-case statement**: The guard clause and then statement.
+#### Switch case fallthrough
 
-Multiple switch case patterns may share the same variable scope if their case
-bodies are empty:
+In a switch statement or expression, multiple cases may share the same body:
 
 ```dart
 switch (obj) {
   case [int a, int b]:
   case {"a": int a, "b": int b}:
     print(a + b); // OK.
+}
+```
 
-  case [int a]:
-  case (String a): // Error.
-    break;
+This would normally be a name collision, but we make an exception to allow it
+when it doesn't cause any problems.
+
+When compiling the body shared by a set of cases with pattern variable sets `vs`
+(where a default case has an empty pattern variable set), the enclosing scope is
+a synthesized scope `s` defined as:
+
+1.  Let `n` be the union of the sets of names of all of the pattern variable
+    sets in `vs`.
+
+2.  For each name in `n`:
+
+    1.  If `n` is defined in every pattern variable set in `vs` and has the same
+        type and finality, then introduce `n` into `s` with that type and
+        finality.
+
+        If any of the corresponding variables in `vs` are promoted, calculate
+        the promoted type of the variable in `s` based on all of the promoted
+        types of `n` in the cases in the same way that promotions are merged at
+        join points.
+
+    2.  Else `n` is inconsistently defined by the cases. Introduce `n` into `s`
+        with a sentinel type `Bad`.
+
+Compile the body using that scope. It is a compile-time error if any identifier
```

Comment:
that scope -> that static scope
(for emphasis).

Reply:
Done.